### ISSUE TITLE
Blockbase/FSE themes - Navigation Block: Site Editor does not accurately show changes made in Display module for nav block

### ISSUE THREAD

**Issue by cuemarie** [Automattic/wp-calypso#65419]
### Quick summary

Changing font size, color, and other Typography styles for the Navigation block results in some changes showing in the Site Editor, and others are only visible on the live site.

### Steps to reproduce

1. Install theme Meraki on a test site
2. Open Site Editor and create a navigation block with a few menu items
3. Under the nav block's block settings, adjust all settings under the `Display` module

### What you expected to happen

Changes made to the menu's color and typography are reflected in the Site Editor

### What actually happened

The following properties appear in the site editor correctlY: 

- Color: Text, Background, and Submenu & overlay background
- Typography: Appearance, Decoration

The following properties are only visible on Submenu menu items:

- Typography: Font family, Size, and Letter case

The last item is not visible at all:

- Color: Submenu & overlay text

<img width="1109" alt="Markup 2022-07-08 at 15 04 28" src="https://user-images.githubusercontent.com/27249804/178088039-ffbf520d-1290-4ec0-8599-50fb8f577787.png">

#### Live Site

On the live site, all changes are visible, except that the `Submenu & overlay text` color is improperly applied to the main menu, and not applied to the submenu.

<img width="1399" alt="Markup 2022-07-08 at 15 06 17" src="https://user-images.githubusercontent.com/27249804/178088044-646ea8cd-9df3-48d8-b099-f0fb67df1e92.png">

### Context

Reported here: 36018994-hc
Customer was trying to change font size, and noting that things looked different between editor and live site

**Follow Up Ticket: 5360428-zd-woothemes**

### Simple, Atomic or both?

Atomic

### Theme-specific issue?

Meraki

### Browser, operating system and other notes

Only had a change to test this on Atomic

I did a quick test on other FSE themes and the issues did not present in the same way

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

Yes, difficult to implement

### Workaround details

- Open up the live site in a new tab/window 
- Return to the Site Editor and adjust the Navigation Block's display settings
- Select **Save** to save that change
- Refresh view of live site to see the results

**Comment by Robertght** [comment]
I'm not moving this yet, but I asked here p1657650535013499-slack-C029FM1EH

**Comment by Addison-Stavlo** [comment]
I can repro on Meraki, but not TT2 - perhaps an issue with Blockbase child themes? cc @pbking @jeffikus

**Comment by madhusudhand** [comment]
Issue can be replicated with any theme that has style definitions for `core/navigation-link` in `theme.json`

This seems to be an issue with Gutenberg itself. Created an issue here https://github.com/WordPress/gutenberg/issues/42963

**Comment by jeffikus** [comment]
Thanks @madhusudhand - I'm removing this from our board.

**Comment by Addison-Stavlo** [comment]
This seems like it might be the same issue as https://github.com/Automattic/wp-calypso/issues/62017 ?

If so we could close one in favor of the other.

**Comment by KirkwallDay** [comment]
- [ ] 5500136-zen another instance here

**Comment by hngdny** [comment]
Another report here 37050063-hc

**Comment by jamiepalatnik** [comment]
Adding that the user from this comment requested a follow-up if there is a resolution, so I created a follow-up ticket: 5578503-zd-woothemes

**Comment by Robertght** [comment]
5579740-zen

**Comment by Robertght** [comment]
5601369-zen Submenu & Overlay Text does not work as set within the editor.

**Comment by filipanoscampos** [comment]
Another case here:
5638928-zd-woothemes

**Comment by kspilarski** [comment]
5830773-zd Varese theme.

**Comment by ha-un** [comment]
Another case: 40059682-hc

**Comment by wiesenhauss** [comment]
Another case here: 40667803-hc

**Comment by NoHopeRadio** [comment]
Another case in 40789792-hc, I provided CSS to undo the `a:where(:not(.wp-element-button))` rules so that the H2 rules took over in a query loop.

```
/*unset style rules in nested links */
a:where(:not(.wp-element-button)) {
    font-size: unset;
    font-weight: unset;
    font-style: unset;
}
```

I don't quite understand how rules for this element seem to take precedence, though.

**Comment by NoHopeRadio** [comment]
Maybe this bug was happening in 40928618-hc in the site title element of the 
Spearhead Blocks theme, though it seemed that user was experiencing some other color anomalies that I couldn't recreate.

**Comment by the-misha** [comment]
Another case: 6352918-zd-woothemes

AT/Pro site, Marl theme. Worked around with CSS:
```
/* set submenu item color | 6352918-zd tdl */
.wp-block-navigation-item__content {
    color: white !important;
    background-color: black;
}
```

**Comment by cuemarie** [comment]
📌 ACTIONS
- Due to the age of this issue, I've downgraded the priority from `High` to `Normal` to better reflect its status in the larger context of open bugs in this repo.
- I also think there are a few issues I tried reporting all in one here, and I think we'd probably be more successful disconnecting them and reporting them in pieces.  I'd also bet that they aren't all happening anymore. 
- I've **marked this for retesting** and I'll try to clear things up after retesting the different pieces of the issue. 👍
